src/core/backbone_router/leader.cpp

```diff
@@ -186,6 +235,62 @@ void Leader::Update(void)
 #endif
 }
 
+void Leader::UpdateDomainPrefixConfig(void)
+{
+    NetworkData::Iterator           iterator = NetworkData::kIteratorInit;
+    NetworkData::OnMeshPrefixConfig config;
+    DomainPrefixState               state;
+    bool                            found = false;
+
+    while (Get<NetworkData::Leader>().GetNextOnMeshPrefix(iterator, config) == OT_ERROR_NONE)
+    {
+        if (config.mDp)
+        {
+            found = true;
+            break;
+        }
+    }
+
+    if (!found)
+    {
+        if (mDomainPrefix.mLength != 0)
+        {
+            // Domain Prefix does not exist any more.
+            mDomainPrefix.mLength = 0;
+            state                 = kDomainPrefixRemoved;
+        }
+        else
+        {
+            state = kDomainPrefixNone;
+        }
+    }
+    else if (config.mPrefix.mLength == mDomainPrefix.mLength &&
+             otIp6PrefixMatch(&mDomainPrefix.mPrefix, &config.mPrefix.mPrefix) >= mDomainPrefix.mLength)
+    {
+        state = kDomainPrefixUnchanged;
+    }
+    else
+    {
+        if (mDomainPrefix.mLength == 0)
+        {
+            state = kDomainPrefixAdded;
+        }
+        else
+        {
+            state = kDomainPrefixRefreshed;
+        }
+
+        mDomainPrefix = config.mPrefix;
+    }
+
+    LogDomainPrefix(state, mDomainPrefix);
+
+    if (state == kDomainPrefixAdded || state == kDomainPrefixRefreshed || state == kDomainPrefixRemoved)
+    {
+        Get<Notifier>().Signal(OT_CHANGED_THREAD_DOMAIN_PREFIX);
+    }
```

Comment:
Do we know if the users of OT would need to be notified on the "Domain Prefix" change?

If not,  can we consider not using the `Notifier` for this?  I see that we currently only monitor this "changed flag" from `Mle` (where it does update `mAllDomainBackboneRouters`). We can consider adding a  method in `Mle` that does that and call it directly from here.

Thoughts?

Reply:
Currently not mandatory to use notification way, changed to direct call as your suggestion.
Please take a review.